test/runtests.jl

```diff
@@ -71,3 +71,39 @@ end
         @test dir1_log == dir2_log
     end; end
 end
+
+# https://github.com/JuliaVersionControl/Git.jl/issues/51
+@testset "OpenSSH integration" begin
+    is_ci = parse(Bool, strip(get(ENV, "CI", "false")))
+    is_gha = parse(Bool, strip(get(ENV, "GITHUB_ACTIONS", "false")))
+    if is_ci && is_gha
+        @info "This is GitHub Actions CI, so running the OpenSSH test..."
+        mktempdir() do sshprivkeydir
+            privkey_filepath = joinpath(sshprivkeydir, "my_private_key")
+            open(privkey_filepath, "w") do io
+                ssh_privkey = ENV["CI_READONLY_DEPLOYKEY_FOR_CI_TESTSUITE_PRIVATEKEY"]
+                println(io, ssh_privkey)
+            end # open
+            # We need to chmod our private key to 600, or SSH will ignore it.
+            chmod(privkey_filepath, 0o600)
+
+            ssh_verbose = "-vvv" # comment this line back out when you are finished debugging
+            # ssh_verbose = "" # uncomment this line when you are finished debugging
```

Comment:
TODO once we have finished debugging CI in this PR:

```suggestion
            # ssh_verbose = "-vvv" # comment this line back out when you are finished debugging
            ssh_verbose = "" # uncomment this line when you are finished debugging
```